### ISSUE TITLE
"FAIL at criu lazy-pages/page-server exited with" with no proper error message visible

### ISSUE THREAD

**Issue by Snorch** [checkpoint-restore/criu#1280]
Reported by Travis https://travis-ci.org/github/checkpoint-restore/criu/jobs/744605516

```
====================== Run zdtm/transition/lazy-thp in ns ======================
Start test
./lazy-thp --pidfile=lazy-thp.pid --outfile=lazy-thp.out
Run criu dump
=[log]=> dump/zdtm/transition/lazy-thp/96/1/dump.log
------------------------ grep Error ------------------------
b'(00.028384)'
b'(00.028445) Collecting mappings (pid: 96)'
b'(00.028451) ----------------------------------------'
b'(00.028699) Dumping path for -3 fd via self 12 [/zdtm/transition/lazy-thp]'
b"(00.028782) Warn  (criu/files-reg.c:1510): Couldn't find the build-id note for file with fd 14"
b'(00.028924) vma 563c2f43f000 borrows vfi from previous 563c2f43b000'
b'(00.028949) vma 563c2f441000 borrows vfi from previous 563c2f43f000'
b'(00.028969) vma 563c2f442000 borrows vfi from previous 563c2f441000'
b'(00.029038) Dumping path for -3 fd via self 12 [/lib/ld-musl-x86_64.so.1]'
b"(00.029071) Warn  (criu/files-reg.c:1510): Couldn't find the build-id note for file with fd 15"
b'(00.036367) Parasite syscall_ip at 0x563c2f43b000'
b'(00.037507) Set up parasite blob using memfd'
b'(00.037538) Putting parasite blob into 0x7f766e88e000->0x7f48f75d8000'
b'(00.037593) Dumping general registers for 99 in native mode'
b'(00.037597) Warn  (compel/arch/x86/src/lib/infect.c:281): Will restore 99 with interrupted system call'
------------------------ ERROR OVER ------------------------
Run criu page-server
Run criu lazy-pages
Run criu restore
Send the 15 signal to  117
Wait for zdtm/transition/lazy-thp(117) to die for 0.100000
=[log]=> dump/zdtm/transition/lazy-thp/96/1/page-server.log
------------------------ grep Error ------------------------
b'(00.001667) Running status-ready scripts'
b'(00.178545) Accepted connection from 127.0.0.1:54844'
b'(00.233720) tls: TLS handshake completed'
b'(00.235191) page-pipe: Created tee pipe size 4194304'
b'(00.263751) Error (criu/tls.c:147): tls: Failed receiving data: Error in the pull function.'
b"(00.263771) Error (criu/page-xfer.c:1225): page-xfer: Can't read pagemap from socket: I/O error"
------------------------ ERROR OVER ------------------------
##### Test zdtm/transition/lazy-thp FAIL at criu page-server exited with 1 #####
Test output: ================================
08:43:01.210:     4: PASS

 <<< ================================

==== ALARM ====
```

or

https://travis-ci.org/github/checkpoint-restore/criu/jobs/742657831
```
======================== Test zdtm/transition/fork PASS ========================
======================== Run zdtm/transition/fork in h =========================
Start test
./fork --pidfile=fork.pid --outfile=fork.out
free(): double free detected in tcache 2
Run criu dump
=[log]=> dump/zdtm/transition/fork/5753/1/dump.log
------------------------ grep Error ------------------------
b'(00.001600) Seized task 5753, state 1'
b'(00.001609) seccomp: Collected tid_real 5753 mode 0'
b'(00.001632) Collected (4 attempts, 0 in_progress)'
b'(00.001658) Seized task 11643, state 0'
b'(00.001669) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11643 (Operation not permitted)'
b'(00.001720) Collected (4 attempts, 0 in_progress)'
b'(00.001734) Collected (4 attempts, 0 in_progress)'
b'(00.001745) Collected 11643 in 2 state'
b'(00.001748) Seized task 11645, state 0'
b'(00.001752) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11645 (Operation not permitted)'
b'(00.001800) Collected (4 attempts, 0 in_progress)'
b'(00.001814) Collected (4 attempts, 0 in_progress)'
b'(00.001820) Collected 11645 in 2 state'
b'(00.001823) Seized task 11646, state 0'
b'(00.001827) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11646 (Operation not permitted)'
b'(00.001871) Collected (4 attempts, 0 in_progress)'
b'(00.001885) Collected (4 attempts, 0 in_progress)'
b'(00.001891) Collected 11646 in 2 state'
b'(00.001894) Seized task 11647, state 0'
b'(00.001897) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11647 (Operation not permitted)'
b'(00.001940) Collected (4 attempts, 0 in_progress)'
b'(00.001953) Collected (4 attempts, 0 in_progress)'
b'(00.001959) Collected 11647 in 2 state'
b'(00.001962) Seized task 11649, state 0'
b'(00.001965) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11649 (Operation not permitted)'
b'(00.002007) Collected (4 attempts, 0 in_progress)'
b'(00.002021) Collected (4 attempts, 0 in_progress)'
b'(00.002028) Collected 11649 in 2 state'
b'(00.002031) Seized task 11650, state 0'
b'(00.002035) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11650 (Operation not permitted)'
b'(00.002078) Collected (4 attempts, 0 in_progress)'
b'(00.002091) Collected (4 attempts, 0 in_progress)'
b'(00.002095) Collected 11650 in 2 state'
b'(00.002097) Seized task 11651, state 0'
b'(00.002101) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11651 (Operation not permitted)'
b'(00.002142) Collected (4 attempts, 0 in_progress)'
b'(00.002154) Collected (4 attempts, 0 in_progress)'
b'(00.002161) Collected 11651 in 2 state'
b'(00.002164) Seized task 11653, state 0'
b'(00.002168) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11653 (Operation not permitted)'
b'(00.002213) Collected (4 attempts, 0 in_progress)'
b'(00.002226) Collected (4 attempts, 0 in_progress)'
b'(00.002236) Collected 11653 in 2 state'
b'(00.002239) Seized task 11654, state 0'
b'(00.002242) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11654 (Operation not permitted)'
b'(00.002291) Collected (4 attempts, 0 in_progress)'
b'(00.002304) Collected (4 attempts, 0 in_progress)'
b'(00.002310) Collected 11654 in 2 state'
b'(00.002313) Seized task 11655, state 0'
b'(00.002317) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11655 (Operation not permitted)'
b'(00.002361) Collected (4 attempts, 0 in_progress)'
b'(00.002374) Collected (4 attempts, 0 in_progress)'
b'(00.002380) Collected 11655 in 2 state'
b'(00.002383) Seized task 11656, state 0'
b'(00.002387) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11656 (Operation not permitted)'
b'(00.002433) Collected (4 attempts, 0 in_progress)'
b'(00.002445) Collected (4 attempts, 0 in_progress)'
b'(00.002451) Collected 11656 in 2 state'
b'(00.002454) Seized task 11657, state 0'
b'(00.002458) Warn  (compel/src/lib/infect.c:127): Unable to interrupt task: 11657 (Operation not permitted)'
------------------------ ERROR OVER ------------------------
Run criu page-server
Run criu lazy-pages
Run criu restore
Send the 15 signal to  5753
Wait for zdtm/transition/fork(5753) to die for 0.100000
####### Test zdtm/transition/fork FAIL at criu lazy-pages exited with -6 #######
Test output: ================================
09:45:02.150:  5753: PASS

 <<< ================================
criu page-server exited with 0

==== ALARM ====
```

Manually I fail to reproduce it with `./test/zdtm.py run -p 2 -t zdtm/transition/fork --remote-lazy-pages`.

I have probably stupid idea - to make grep_errors() in test/zdtm.py also print at least last 10 lines of the log if there are no errors in log but ret != 0. -> Will do tomorrow if nobody does it faster.

**Comment by adrianreber** [comment]
See #1277 for a 'lazy' workaround.

**Comment by rppt** [comment]
I have no clue where `-6 (ENXIO)` can come from in `criu lazy-pages exited with -6` except page-xfer. Could be a TLS again?

> Manually I fail to reproduce it with `./test/zdtm.py run -p 2 -t zdtm/transition/fork --remote-lazy-pages`

Usually to reproduce corner cases with `lazy-pages` I had to run some stress in the background and `zdtm` in loop ;-)

**Comment by Snorch** [comment]
JFYI: https://api.travis-ci.org/v3/job/744854232/log.txt
```
=[log]=> dump/zdtm/transition/fork/2210/1/lazy-pages.log
------------------------ grep Error ------------------------
(00.139986) uffd: 4-12: #PF at 0x7ffc5944e000
(00.139989) uffd: 4-12: zero page at 0x7ffc5944e000
(00.140696) uffd: 4-19: uffd_copy: 0x7ffc5944f000/4096
(00.140705) uffd: 4-19: EXIT
(00.140727) Error (criu/uffd.c:1220): uffd: 4--19: Can't read uffd message: Bad file descriptor
------------------------ ERROR OVER ------------------------
####### Test zdtm/transition/fork FAIL at criu lazy-pages exited with 1 ########
```

**Comment by rppt** [comment]
> JFYI: https://api.travis-ci.org/v3/job/744854232/log.txt
> 
```
=[log]=> dump/zdtm/transition/fork/2210/1/lazy-pages.log
------------------------ grep Error ------------------------
(00.139986) uffd: 4-12: #PF at 0x7ffc5944e000
(00.139989) uffd: 4-12: zero page at 0x7ffc5944e000
(00.140696) uffd: 4-19: uffd_copy: 0x7ffc5944f000/4096
(00.140705) uffd: 4-19: EXIT
(00.140727) Error (criu/uffd.c:1220): uffd: 4--19: Can't read uffd message: Bad file descriptor
------------------------ ERROR OVER ------------------------
####### Test zdtm/transition/fork FAIL at criu lazy-pages exited with 1 ########
```
> 
> Likely can be fixed by [#1277 (comment)](https://github.com/checkpoint-restore/criu/pull/1277#issuecomment-730520228). @rppt

Yeah, I think so. I'll rebase that patch and will send a PR.

**Comment by Snorch** [comment]
Let's close this after  #1286
